py/selenium/webdriver/support/expected_conditions.py

```diff
@@ -95,6 +95,8 @@ def __call__(self, ignored):
 
 
 def _element_if_visible(element, visibility=True):
+    if isinstance(element, str) or isinstance(element, dict):
+           raise StaleElementReferenceException("Invalid locator")
```

Comment:
This is not the right exception class. There is an InvalidSelectorException class that covers bad locators.